### PR TITLE
AHOYAPPS-948: Add recording indicator

### PR DESCRIPTION
**Pull request by timrozum**
https://issues.corp.twilio.com/browse/AHOYAPPS-948

#### Changes

1. Add recording indicator to room screen, with glow animation!

#### Testing

1. Manually tested using a web deployment that has feature to start and stop recording.
2. No UI tests because we can't control recording from the iOS app and so it would probably be a large effort.
3. Did not add unit tests for a variety of reasons. I don't think `VideoApp.Room` will be able to mock `TwilioVideo.Room` very well because I can't create a `TwilioVideo.Room` to pass to the delegate functions. Less mocking problems with `RoomViewController` and `RoomViewModel` but these are kind of better suited to UI tests. `ParticipantsStore` would make sense for unit tests because it's not UI and there are no mocking problems but the change was so trivial and this file does not yet have a test spec. I'm not opposed to adding unit tests to `ParticipantStore` for this but it kind of seemed best to save that for more interesting changes or tackle the entire file with a separate ticket and move on to other things for now.

Screen Shot 2021-01-13 at 10 23 06 AM

Screen Shot 2021-01-13 at 10 22 46 AM

https://user-images.githubusercontent.com/1930363/104494306-841d5380-5593-11eb-8f2f-73cee6f67d83.mov

### CODE REVIEW

File: VideoApp/VideoApp/Views/RecordingDotView/RecordingDotView.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    
+    private func brighten() {
+        UIView.animate(
+            withDuration: 0.6,
+            delay: .zero,
+            options: .curveEaseOut,
+            animations: { self.backgroundColor = .red },
+            completion: { [weak self] _ in self?.darken() }
+        )
+    }
```
Comment (timrozum) [inline VideoApp/VideoApp/Views/RecordingDotView/RecordingDotView.swift:44]: I looked into using `animateKeyFrames` to group these but the API does not provide a simple way to ease out. It seems it has to be done manually.

But this looks really good and code isn't too bad.

File: Podfile.lock [hunk 2]
```diff
@@ -108,9 +108,9 @@ PODS:
   - nanopb/decode (1.30906.0)
   - nanopb/encode (1.30906.0)
   - Nimble (9.0.0)
-  - PromisesObjC (1.2.11)
+  - PromisesObjC (1.2.12)
   - Quick (3.0.0)
-  - TwilioVideo (4.0.0)
+  - TwilioVideo (4.0.1)
```
Comment (paynerc) [inline Podfile.lock:113]: `4.1.0` is out these days. Just sayin'....

File: VideoApp/VideoApp/Views/RecordingDotView/RecordingDotView.swift (hunk 1 above)
Comment (paynerc) [reply VideoApp/VideoApp/Views/RecordingDotView/RecordingDotView.swift:44]: Cool stuff!!

**Review by paynerc — APPROVED** [review]
I like it!

**Pull request merged**